examples/decomposition/plot_kernel_pca_approximate.py

```diff
@@ -0,0 +1,128 @@
+top row show approximations of this data made by KernelPCA transform + inverse
+transform, for various number of components. The bottom row shows the data
+samples projected along the first two principal components for each case.
+
+What you can observe:
+---------------------
+With quite small `n_components`, the approximation is almost identical to the
+full KernelPCA using 2000 components, while execution is much faster
+(execution times are included in the plot titles)
+
+Going further:
+--------------
+You can have a look at the `Kernel PCA Solvers comparison benchmarks
+<https://scikit-learn.org/stable/modules/decomposition.html#kernel-pca-solvers-benchmarks>`_
+to compare execution times in more details.
+"""
+print(__doc__)  # noqa
+
+# Authors: Sylvain MARIE, Schneider Electric
+# License: BSD 3 clause
+
+from datetime import datetime
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.decomposition import KernelPCA
+from sklearn.datasets import make_circles
+
+
+# 1- Generate random data
+# -----------------------
+np.random.seed(0)
+n_samples = 2000
+X, y = make_circles(n_samples=n_samples, factor=.3, noise=.05)
+
+
+# 2- Design experiment / Init plots
+# ---------------------------------
+plt.ion()
+plt.figure(figsize=(11, 5.5))
+
+# Set this to true to create a logarithmic grid
+use_log_grid = False
+if use_log_grid:
+    grid_size = 4
+    n_components_range = [np.round(np.exp((x / grid_size) * np.log(n_samples)))
+                          for x in range(1, grid_size + 1)]
+else:
+    # hardcoded for the example gallery
+    n_components_range = [4, 40, 2000]
+
+# top left: original
+nb_cols = 1 + len(n_components_range)
+plt.subplot(2, nb_cols, 1, aspect='equal')
+plt.title("Original space")
+reds = y == 0
+blues = y == 1
+
+plt.scatter(X[reds, 0], X[reds, 1], c="red",
+            s=20, edgecolor='k')
+plt.scatter(X[blues, 0], X[blues, 1], c="blue",
+            s=20, edgecolor='k')
+plt.xlabel("$x_1$")
+plt.ylabel("$x_2$")
+plt.draw()
+plt.pause(1e-3)  # needed to refresh the display
+
+# 2- Run experiments
+# -----------------------
+for i, n_compo in enumerate(reversed(n_components_range)):
+    n_compo = int(n_compo)
+    print("Fitting kPCA for n_components=%i..." % n_compo)
+    start_time = datetime.now()
+
+    # fit and transform
+    kpca = KernelPCA(n_components=n_compo, kernel="rbf",
+                     fit_inverse_transform=True, gamma=10)
+    X_kpca = kpca.fit_transform(X)
+    X_back = kpca.inverse_transform(X_kpca)
+    elapsed = (datetime.now() - start_time).total_seconds()
+    print("...done. Elapsed time: %.2fs" % elapsed)
+
+    # original space after inverse transform
+    plt.subplot(2, nb_cols, i + 2, aspect='equal')
+    plt.scatter(X_back[reds, 0], X_back[reds, 1], c="red",
+                s=20, edgecolor='k')
+    plt.scatter(X_back[blues, 0], X_back[blues, 1], c="blue",
+                s=20, edgecolor='k')
```

Comment:
The first row is always the same, right? Why not have a single row where teh original data is the leftmost plot?

Reply:
All figures in the plots correspond to a different computation, so I am not sure I understand your comment.